packages/statemanager/src/stateManager.ts

```diff
@@ -235,6 +236,11 @@ export class DefaultStateManager implements EVMStateManagerInterface {
         type: this._storageCacheSettings.type,
       })
     }
+
+    this._codeCache2 = new CodeCache({
+      size: 20000,
+      type: CacheType.ORDERED_MAP,
```

Comment:
Type is always a `ORDERED_MAP`, do we need LRU cache support in code cache then?

Reply:
It's the same with the account and storage caches too. Although the value is defaulted, the cache type option is given to the user. Furthermore, all of the caches will be refactored and unified in a future PR, so I'm not opposed to dropping support and only using e.g. `ORDERED_MAP` if it outperforms `LRU` in most/all relevant use cases.

Reply:
These are two datastructures with different properties, we need both, also for the code cache (so this should be added here as well).

`ORDERED_MAP` performs somewhat better, but is not bound in size, it also works better (at all) for smaller caches. `LRU_CACHE` manages the cache size and drops out older items and the like (this is where the overhead is coming from). We use `LRU` in the client, since if we would not caches would grow indefinitely and the client will go out of memory. `ORDERED_MAP` in return is used in short-lived contexts like the VM blockchain and state tests.

Also just a a word of caution: I have "unified" state and account cache as much as I ever could when I reworked the caches early this year, so many of the design decisions are relatively strongly grounded. Doesn't mean things can not be improved likely, but let's please align closely here before new refactorings of the caches are started. Best to open one or some issues on that clearly describing what you plan to do.

Reply:
So code cache should also get an LRU option and the client should instantiate the cache with this option and then there should be also an additional flag `codeCache` for the client analog to e.g. `accountCache` or `storageCache`.

Reply:
Regarding the refactor of the cache: I think both storage/account (and now code) cache look rather similar, so if we could combine them in a base class and then further pick out any differences for the specific caches then this means we could re-use the cache mechanism in the future if we need a new cache somewhere :smile:

Regarding `LRU_CACHE`: ah great, OK we definitely need this for client.